src/SessionSetup/SessionSetupUtils.cpp

```diff
@@ -27,4 +29,21 @@ std::shared_ptr<grpc::Channel> CreateGrpcChannel(uint16_t port) {
   return result;
 }
 
+std::optional<std::pair<QString, uint32_t>> SplitInstanceAndProcessIdFromConnectionTarget(
```

Comment:
This code is all good but nevertheless I wanted to make a suggestion that I believe is more C++-y. So consider this as a Nit and feel free to ignore:

I think in C++ you would rather centralize such logic around a type. That would mean to create a class `ConnectionTarget` with a private constructor and a static factory function which incorporates the logic of this function. Something like that (plus nodiscard and so on):

```
class ConnectionTarget {
public:
  static std::optional<ConnectionTarget> FromString(const QString&);

  uint32_t proccess_id() const;
  QString instance( ) const;

  friend bool operator==(const ConnectionTarget&, const ConnectionTarget&);
  friend bool operator!=(const ConnectionTarget&, const ConnectionTarget&);

private:
  // ...
};

Reply:
Not sure I agree that it's really more C++-y. It's certainly more Object-Oriented, but I think it's overblown given that I really just want a utility method to split a string into two parts, and the only reason I didn't simply add this in "main.cpp" is that I wanted to write a test for it - I think there's a place for free functions in C++ as well.

I might be convinced that it would be clearer to return a `struct { QString, uint32_t }` instead of the `std::pair<>` though...

Reply:
Well this is about a value, not an object, so this is according to my understanding not object oriented at all. And yes, returning a struct from a free function is basically the same what I suggested, just without the enforcement of invariance.

But like I said, it's a nit. Feel free to ignore.

Reply:
I've changed this to return a struct, I agree that this is cleaner. WDYT?

Reply:
Awesome. LGTM